src/imageio/imageio_rgbe.c

```diff
@@ -414,25 +414,31 @@ dt_imageio_retval_t dt_imageio_open_rgbe(dt_image_t *img, const char *filename,
   rgbe_header_info info;
   if(RGBE_ReadHeader(f, &img->width, &img->height, &info)) goto error_corrupt;
 
+  float *readbuf = g_malloc((size_t)img->height * img->width * 3 * 4);
+  if(!readbuf) goto error_cache_full;
+
   img->buf_dsc.channels = 4;
   img->buf_dsc.datatype = TYPE_FLOAT;
-  float *buf = (float *)dt_mipmap_cache_alloc(mbuf, img);
-  if(!buf) goto error_cache_full;
+  float *mipbuf = (float *)dt_mipmap_cache_alloc(mbuf, img);
+  if(!mipbuf) goto error_cache_full;
 
-  if(RGBE_ReadPixels_RLE(f, buf, img->width, img->height)) goto error_corrupt;
+  if(RGBE_ReadPixels_RLE(f, readbuf, img->width, img->height)) goto error_corrupt;
   fclose(f);
 
   // repair nan/inf etc
   const size_t width = img->width;
   const size_t height = img->height;
 #ifdef _OPENMP
 #pragma omp parallel for default(none) \
-  dt_omp_firstprivate(width, height, buf) \
-  collapse(2)
+  dt_omp_firstprivate(width, height, readbuf) \
+  shared(mipbuf)
```

Comment:
Since the loop isn't modifying the `mipbuf` pointer itself, this can be a firstprivate variable as well.  Depending on what other optimizations the compiler performs (or has to skip due to it being `shared`), that can speed up the code.

Reply:
@ralfbrown Thank you for reviewing. And since the alternative PR is better, I'll close this one.